l5kit/l5kit/data/zarr_utils.py

```diff
@@ -1,66 +1,223 @@
+        except (ValueError, KeyError):
+            print(f"{input_zarr} is not valid! skipping")
+            continue
+        num_els_valid_zarrs.append(_get_num_els_in_scene_range(input_dataset, 0, len(input_dataset.scenes)))
+        valid_zarrs.append(input_zarr)
+
+    # we can now pre-allocate the output dataset
+    total_num_els: Counter = Counter()
+    for num_el in num_els_valid_zarrs:
+        total_num_els += Counter(num_el)
+    output_dataset.initialize(**total_num_els)
+
+    cur_num_els = Counter({"num_scenes": 0, "num_frames": 0, "num_agents": 0, "num_tl_faces": 0})
+    tqdm_bar = tqdm(valid_zarrs)
+    for idx, input_zarr in enumerate(tqdm_bar):
+        tqdm_bar.set_description(f"working on {input_zarr}")
 
         input_dataset = ChunkedDataset(input_zarr)
         input_dataset.open()
 
-        if verbose:
-            print(f"input scenes size: {len(input_dataset.scenes)}")
-            print(f"input frames size: {len(input_dataset.frames)}")
-            print(f"input agents size: {len(input_dataset.agents)}")
-            print(f"input tl_faces size: {len(input_dataset.tl_faces)}")
-
-        frame_offset = len(output_dataset.frames)
-        new_scenes = np.zeros(len(input_dataset.scenes), dtype=SCENE_DTYPE)
-
-        for i, scene in enumerate(input_dataset.scenes):  # add new scenes to zarr
-            scene["frame_index_interval"] = scene["frame_index_interval"] + frame_offset
-            new_scenes[i] = scene
-        output_dataset.scenes.append(new_scenes)
-
-        agent_offset = len(output_dataset.agents)
-        tl_faces_offset = len(output_dataset.tl_faces)
-
-        new_frames = np.zeros(len(input_dataset.frames), dtype=FRAME_DTYPE)
-        for i, frame in enumerate(input_dataset.frames):  # add new frames to the zarr
-            frame["agent_index_interval"] = frame["agent_index_interval"] + agent_offset
-            frame["traffic_light_faces_index_interval"] = frame["traffic_light_faces_index_interval"] + tl_faces_offset
-            new_frames[i] = frame
-        output_dataset.frames.append(new_frames)
-
-        output_dataset.agents.append(input_dataset.agents)  # add new agents to the zarr
-        output_dataset.tl_faces.append(input_dataset.tl_faces)  # add new traffic light faces to the zarr
-
-    if verbose:
-        print(f"output scenes size: {len(output_dataset.scenes)}")
-        print(f"output frames size: {len(output_dataset.frames)}")
-        print(f"output agents size: {len(output_dataset.agents)}")
-        print(f"output tl_faces size: {len(output_dataset.tl_faces)}")
+        _append_zarr_subset(input_dataset, output_dataset, 0, len(input_dataset.scenes), cur_num_els)
+        cur_num_els += Counter(num_els_valid_zarrs[idx])
+
+
+def zarr_split(input_zarr: str, output_path: str, split_infos: List[dict]) -> List[Tuple[int, int]]:
+    """
+    Split the input zarr into many zarrs. Names and sizes can be passed using the split_infos arg.
+
+    Args:
```

Comment:
Minor, nice to have but not sure if worth it: overload this function to work with only a single argument for size denoting the number / size of each split.

Reply:
mmmmm python doesn't support function overloading easily so I would leave it for another PR maybe